Improve handling for Elastic IPs in EKS and NAT modules

ahorner opened the pull request:
As I worked through the process of spinning up an EKS cluster for one of our clients, I noticed that there were a few missing inputs/outputs necessary to join some of these resources together in the happy-path case (all resources are contained within a single AWS account).

This exposes the Elastic IP for our NAT gateway module, and _optionally_ exposes it for our EKS module.

ahorner committed:
Expose elastic_ip as an output for EKS module

This is useful for targeting the cluster's NAT gateway in security
rules for resources external to the cluster's VPC.

ahorner committed:
Support ingress CIDR blocks for ElastiCache

To assist in cross-VPC communication, we need to be able to include
CIDR block entries for, e.g., Elastic IP addresses. Our RDS module
already supports this.

vandrijevik commented on aws/eks/outputs.tf:
```diff
@@ -25,3 +25,7 @@ output "subnets" {
 }
+
+output "elastic_ip" {
+  value = var.uses_nat_gateway ? module.eks-vpc-nat-gateway[0].elastic_ip : ""
```
Rather than `""` here, can we return `null` when the output is undefined?

vandrijevik commented on aws/elasticache/main.tf:
```diff
@@ -85,6 +85,7 @@ resource "aws_security_group" "sg_on_elasticache_instance" {
     from_port       = local.port
     to_port         = local.port
     protocol        = "tcp"
+    cidr_blocks     = var.sg_cidr_blocks
```
I think that, as we discussed, and as @rdunlop also discovered, this doesn’t work for Elasticache. Should we remove it from this PR?

ahorner committed:
Clean up feedback items

- Remove CIDR block variable from Elasticache module
- Return `null` in favor of `""` for non-existent elastic_ips

vandrijevik reviewed (approved):
🎉

ahorner committed:
Merge branch 'main' into expose_eks_nat_elastic_ips
